### PR TITLE
[OV JS] Add readModel(model:str, weights: Tensor) sync&async

### PR DESCRIPTION
**Pull request by almilosz**
### Details:
 - Add `readModel(model: string, weights: Tensor): Promise<Model>` and `readModelSync(model: string, weights: Tensor): Model`
 - Improve error reporting and parameter validation for readModel
 - Fix wrapping up Model in readModel(); on js side instanceof always operator gave false 
 - Change static ModelWrap::wrap method to function similar to other functions for conversion

### Tickets:
 - *CVS-134817*

### CODE REVIEW

File: src/bindings/js/node/tests/read_model.test.js [hunk 1]
```diff
@@ -34,7 +36,16 @@ describe('Core.readModeSync', () => {
     )
   });
 
-  it('readModeSync(modelUint8ArrayBuffer, weightsUint8ArrayBuffer) ', () => {
+  it('readModelSync(model string, weightsTensor) ', () => {
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/read_model.test.js:39]:
```suggestion
  it('readModelSync(modelString, weightsTensor) ', () => {
```

File: src/bindings/js/node/tests/read_model.test.js [hunk 2]
```diff
@@ -55,7 +66,16 @@ describe('Core.readModel', () => {
     assert.equal(model.inputs.length, 1);
   });
 
-  it('readMode(modelUint8ArrayBuffer, weightsUint8ArrayBuffer) ', async () => {
+  it('readModel(model string, weightsTensor) ', async () => {
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/read_model.test.js:69]:
```suggestion
  it('readModel(modelString, weightsTensor) ', async () => {
```

File: src/bindings/js/node/src/core_wrap.cpp [hunk 3]
```diff
@@ -85,11 +85,13 @@ Napi::Value CoreWrap::read_model_sync(const Napi::CallbackInfo& info) {
             model = _core.read_model(model_str, weight_tensor);
         } else if (ov::js::validate<Napi::String>(info, allowed_signatures)) {
             model = _core.read_model(info[0].ToString());
+        } else if (ov::js::validate<Napi::String, TensorWrap>(info, allowed_signatures)) {
+            model = _core.read_model(info[0].ToString(), cast_to_tensor(info, 1));
         } else {
             OPENVINO_THROW("'readModelSync'", ov::js::get_parameters_error_msg(info, allowed_signatures));
         }
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/src/core_wrap.cpp:92]: Probably, it makes sense to use `ReadModelArgs` here.
(I prefer see validation parameters here, but `ReadModelArgs` reduce code in async case. Here could be an exception to do not duplicate logic in two types of readModel methods)

**Review by vishniakov-nikolai — APPROVED** [review]

File: src/bindings/js/node/src/core_wrap.cpp (hunk 3 above)
Comment (almilosz) [reply src/bindings/js/node/src/core_wrap.cpp:92]: `ReadModelArgs` and other structs are marked in *146325* to be refactored

**Pull request merged**